crates/query-engine/translation/src/translation/query/root.rs

```diff
@@ -107,6 +107,62 @@ pub enum ReturnsFields {
 }
 
 /// Translate rows part of query to sql ast.
+///
+/// To improve performance, we want to reduce the rows in scope as much as possible before performing any joins for relationships
+/// To do this, we add a subquery with the where clause, any required joins, and possibly an order by and limit clause
+/// If ordering by columns in another table, we instead add the order by, required joins, and limit to the parent query
+/// If ordering in the subquery, we must also apply the order by in the parent, as ordering is not guaranteed after joins
+///
+/// Example: Order by a column on the target table
+/// ```sql
+/// SELECT
+/// "%1_Album"."Title" AS "Title"
+/// FROM
+///   (
+///     SELECT
+///       "%0_Album".*
+///     FROM
+///       "public"."Album" AS "%0_Album"
+///     ORDER BY
+///       "%0_Album"."AlbumId" ASC
+///     LIMIT
+///       5
+///   ) AS "%1_Album"
+/// ORDER BY
+///   "%1_Album"."AlbumId" ASC
+/// ```
+///
+/// Example: Order by a related column
+/// ```sql
+/// SELECT
+///   "%1_Album"."Title" AS "Name"
+/// FROM
+///   (
+///     SELECT
+///       "%0_Album".*
+///     FROM
+///       "public"."Album" AS "%0_Album"
+///   ) AS "%1_Album"
+///   LEFT OUTER JOIN LATERAL (
+///     SELECT
+///       "%2_ORDER_PART_Artist"."Name" AS "Name"
+///     FROM
+///       (
+///         SELECT
+///           "%2_ORDER_PART_Artist"."Name" AS "Name"
+///         FROM
+///           "public"."Artist" AS "%2_ORDER_PART_Artist"
+///         WHERE
+///           (
+///             "%1_Album"."ArtistId" = "%2_ORDER_PART_Artist"."ArtistId"
+///           )
+///       ) AS "%2_ORDER_PART_Artist"
+///   ) AS "%3_ORDER_FOR_Album" ON ('true')
+/// ORDER BY
+///   "%3_ORDER_FOR_Album"."Name" ASC
+/// LIMIT
+///   5 OFFSET 3
+/// ```
 fn translate_rows(
```

Comment:
One way to make this function more digestible is splitting its contents into two functions that reflect the two piece of query structure it is generating:
- One that creates the inner subquery
- One that wraps the subquery returned by the previous function with the outer query